### DISCUSSION TITLE
Setting up the prettier config

### DISCUSSION THREAD

**Discussion by igmtdev** [zed-industries/zed#19814] [Feature Requests]
I would like to set up the Prettier config inside the Zed settings like this so I won’t have to create a .prettierrc for each of my React projects. Is this possible? If this kind of Prettier setup in the Zed settings is possible, how can I set up the Prettier plugins like '@ianvs/prettier-plugin-sort-imports' for sorting imports and 'prettier-plugin-tailwindcss' for ordering Tailwind classes?

![image](https://github.com/user-attachments/assets/1bd10927-b145-4ee9-8b58-916092eaeca0)

Here's my prettier config

```
{
  "arrowParens": "avoid",
  "bracketSpacing": true,
  "endOfLine": "lf",
  "jsxSingleQuote": true,
  "semi": false,
  "singleQuote": true,
  "tabWidth": 2,
  "trailingComma": "none",
  "plugins": [
    "@ianvs/prettier-plugin-sort-imports",
    "prettier-plugin-tailwindcss"
  ],
  "importOrder": [
    "^react$",
    "",
    "^next",
    "",
    "<THIRD_PARTY_MODULES>",
    "",
    "^@/features/(.*)$",
    "",
    "^@/components/(.*)$",
    "",
    "^@/lib/(.*)$",
    "",
    "^@/hooks/(.*)$",
    "",
    "^@/public/(.*)$",
    "",
    "^[./]"
  ],
  "importOrderSeparation": true,
  "importOrderSortSpecifiers": true
}
```

**Comment by igmtdev** [comment]
I tried the Prettier setup like the one in the screenshot I provided above, and it worked; my TSX files were formatted. However, when I installed a package like '@ianvs/prettier-plugin-sort-imports,' my Prettier setup in the Zed settings stopped working. I don’t know if this is a bug, but I can no longer format my other TSX files.

**Reply by edo-zhou** [reply]
https://github.com/zed-industries/zed/discussions/13602#discussioncomment-12877080

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
